src/main/java/com/diffplug/gradle/oomph/OomphIdePlugin.java

```diff
@@ -195,6 +202,9 @@ protected void applyOnce(Project project) {
 				if (!extension.workspaceExists()) {
 					ide.dependsOn(ideSetupWorkspace);
 				}
```

Comment:
The `!extension.workspaceExists()` means that the workspace only gets written for a "clean" run of ide - after that it just uses whatever is there.  That's the right choice for the workspace (don't wanna mess up the user's changes in the IDE), but it looks like `ideSetupInstallation` does not work that way, and is running on every single invocation of `ide`.  Are we sure that is what we want?

Reply:
I could add any check. What would you suggest?

Reply:
because at this point in timeline, when ideSetupInstallation I think the idePath already exists, so I think we need a criteria, which matches better to do things only once.

Reply:
do you have an idea what I could use to check to do it only once?

Reply:
One solution is to have the `installation` stuff not be a separate task, and instead have it run in `ideSetupP2`, around here:

https://github.com/diffplug/goomph/blob/3a5b49b64cc3136827ec0e3d2fc2743c8d80101b/src/main/java/com/diffplug/gradle/oomph/OomphIdeExtension.java#L382-L388

The deeper problem I'm starting to see upon a closer inspection of the new docs, is that the instructions for `How do I find and set a ide setting / preference?` are the same as `How do I find and set a workspace setting / preference?`, but it's not clear what the difference between an "IDE setting" and a "workspace setting" is.

After a little googling, I think the install directory is meant to be immutable - changing settings in the IDE should only change the workspace, not the install directory.  [This article](https://www-01.ibm.com/support/docview.wss?uid=swg21653193) seems to indicate that the proxy settings that you're interested in are stored in the workspace, not the install directory.

These docs could be out of date - if your experience is that changing things and saving them causes a change in the `build/oomph-ide.app`, then my paragraph above is wrong, and we still need this PR to get the functionality you need.  In that case, I would recommend putting the installation stuff into the ideSetupP2 method, as linked above.